### ISSUE TITLE
Stack deploy intermittently fails with 'starting container failed: No such network'

### ISSUE THREAD

**Issue by Mobe91** [moby/moby#35544]
**Description**

Task creation during swarm stack deployment intermittently fails due to `starting container failed: No such network`. This looks like some kind concurrency issue since it happens quite rarely. The task creation will continuously fail in this case and the service needs to be removed and recreated in order to resolve the problem.
It also happened to me when I rebooted a worker node that was running a service task. When the node went back up, the task creation failed with the same error.
The problem has also been mentioned in the [forum](https://forums.docker.com/t/getting-no-such-network-errors-starting-a-stack-in-a-swarm/41202).

**Output of `docker version`:**

```
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:39:28 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:45:38 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 11
 Running: 2
 Paused: 0
 Stopped: 9
Images: 4
Server Version: 17.09.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: awslogs
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: eeb03g61tjpa8s183cwiousib
 Is Manager: false
 Node Address: 172.31.0.232
 Manager Addresses:
  172.31.4.42:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.49-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 990.9MiB
Name: ip-172-31-0-232.eu-central-1.compute.internal
ID: 7VFX:Q7OH:JHPW:NXY7:4EKL:Y6JV:57TF:AWM3:Q33D:3PFO:ZSRQ:SVP6
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 165
 Goroutines: 222
 System Time: 2017-11-19T18:42:44.721103237Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Labels:
 os=linux
 region=eu-central-1
 availability_zone=eu-central-1a
 instance_type=t2.micro
 node_type=worker
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```
My swarm is running on AWS.

**Comment by websitesca** [comment]
Same problem, it happens when we reboot EC2 instances on aws. Then swarm will not put any containers on a particular node, as if that node were bad somehow. But after I check the node, it seems ok, its in the swarm and if I scale a service, now swarm will put containers on that node. Note that sometimes the bad node is the manager and sometimes its a worker.

docker version
```
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:18 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:56 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

docker info
```
Containers: 53
 Running: 12
 Paused: 0
 Stopped: 41
Images: 19
Server Version: 17.09.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: awslogs
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: 42vv9816lo4f9nyzxlvs3iony
 Is Manager: true
 ClusterID: 7cguuakf1xyjsm6zb27uddvl5
 Managers: 1
 Nodes: 3
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 10.0.1.10
 Manager Addresses:
  10.0.1.10:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-1041-aws
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.67GiB
Name: ip-10-0-1-10
ID:xxxxxxxx
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: xxxxxxxx
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Comment by u007** [comment]
same here, with debian jessie manager

**Comment by thaJeztah** [comment]
Possibly related to https://github.com/moby/moby/pull/35634

**Comment by nvrnight** [comment]
I get the same intermittent error on nodes that have never been restarted using VirtualBox, 3 Ubuntu 17.10 VMs, one manager, two workers. From the outside looking in it seems like a race condition creating the network. Sometimes the network is not getting created in time and the services are trying to get created before the network is created/available and causes a failure.

**Comment by Mobe91** [comment]
@nvrnight What docker version are you using? This seems to have been fixed in 17.12. I am on 18.03 now and I have not seen this error for a long time.

I am closing this issue since I regard it as fixed.
